backend/src/game_objects/grid.rs

```diff
@@ -6,6 +6,27 @@ pub struct Grid {
     pub hexes: Vec<Hex>,
 }
 
+impl Grid {
+    pub fn new(row_n: u32, col_n: u32) -> Grid {
+        let hexes = (0..row_n).fold(vec![], |mut acc, x| {
```

Comment:
Why not just
```rust
    let mut hexes: Vec<Hex> = vec![];
    for i in 0..row_n {
        for j in 0..col_n {
            hexes.push(Hex { x: i, y: j, unit: None, content: None });
        }
    }
```
?

Reply:
what are this advantages of functional programming for if you can just use two cycles? 

i tired of golang were for .. in is the only option

obviously, i'll fix this

Reply:
For this case, there's none, both approaches are fully equivalent.
However, for more complex ones, there may be many advantages in functional approach.

Also, if you would like to keep things functional, consider avoiding `mut` variables and mutating operations like `append`. I think, more idiomatic functional variant would be something like this:
```rust
let m : Vec<Hex> = (0..row_n).flat_map(
        move |i| (0..col_n).map(
            move |j| {
                Hex { x: i, y: j, unit: None, content: None }
            }).collect::<Vec<Hex>>()
    ).collect();
```

Reply:
Also, in your approach, `acc` is `mut` needlessly. For the `fold` function, the new acc is its return value, so the correct way would be something like this:
```
let hexes = (0..row_n).fold(
    vec![],
    |acc, i| [acc, (0..col_n).map(
        |j| Hex { x: i, y: j, unit: None, content: None }
    ).collect()].concat());
```